spyder_notebook/widgets/client.py

```diff
@@ -115,7 +115,9 @@ class NotebookClient(QWidget):
     def __init__(self, plugin, name):
         super(NotebookClient, self).__init__(plugin)
 
-        self.name = name
+        if os.name == 'nt':
+            name = name.replace('/','\\')
+        self.filename = name
```

Comment:
But to be consistent, you should change `name` to `filename` in `create_new_client`